src/main/flight/pid.c

```diff
@@ -1299,44 +1322,45 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, const rollAndPitchT
             if (cmpTimeUs(currentTimeUs, levelModeStartTimeUs) > CRASH_RECOVERY_DETECTION_DELAY_US) {
                 detectAndSetCrashRecovery(pidProfile->crash_recovery, axis, currentTimeUs, delta, errorRate);
             }
-#if defined(USE_D_CUT)
-            if (dtermCutPercent){
-                dtermCutFactor = dtermCutRangeApplyFn((filter_t *) &dtermCutRange[axis], delta);
-                dtermCutFactor = fabsf(dtermCutFactor) * dtermCutGain;
-                dtermCutFactor = dtermCutLowpassApplyFn((filter_t *) &dtermCutLowpass[axis], dtermCutFactor);
-                dtermCutFactor = MIN(dtermCutFactor, 1.0f);
-                dtermCutFactor = dtermCutPercentInv + (dtermCutFactor * dtermCutPercent);
+#if defined(USE_D_MIN)
+            if (dMinPercentRoll || dMinPercentPitch) {
+                dMinGyroFactor = dMinRangeApplyFn((filter_t *) &dMinRange[axis], delta);
+                dMinGyroFactor = fabsf(dMinGyroFactor) * dMinGyroGain;
+                dMinSetpointFactor = (fabsf(pidSetpointDelta)) * dMinSetpointGain;
+                dMinFactor = MAX(dMinGyroFactor, dMinSetpointFactor);
+                if (axis == FD_ROLL) {
+                    dMinFactor = dMinPercentRoll + (1.0f - dMinPercentRoll) * dMinFactor;
+                } else if (axis == FD_PITCH) {
+                    dMinFactor = dMinPercentPitch + (1.0f - dMinPercentPitch) * dMinFactor;
+                }
```

Comment:
Seems like it would be better to store the `dMinPercent` values as an array and then reference them by `dMinPercent[axis]`. Then this section and other parts could be significantly simplified.

Also, do we need to calculate the filters for an axis if the `dMinPercent` for that axis is 0?

Reply:
Yes that would be great!  I'd be happy to do that.  I wanted the if... out of the pId loop main code too.  I just wasn't sure how to do it.  I'll have a go, but might need some help.

Reply:
I've had a go at it, hopefully I've done OK with it.

Reply:
Thanks for your advice, eTracer, much better now